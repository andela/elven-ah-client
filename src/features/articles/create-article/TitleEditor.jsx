import React from 'react';
import { Editor } from '@tinymce/tinymce-react';
import PropTypes from 'prop-types';

const TitleEditor = ({
  handleChange, handlePaste, value,
  handleTitlePlaceholderTextFocusIn, handleTitlePlaceholderTextFocusOut,
}) => (
  <Editor
    apiKey="3eytfdm1nnstcd6m2hzat9gpfvgg0ik5im0he4tpolgyqne9"
    value={value}
    onEditorChange={handleChange}
    onPaste={handlePaste}
    onFocusIn={handleTitlePlaceholderTextFocusIn}
    onFocusOut={handleTitlePlaceholderTextFocusOut}
    name="title-editor"
    tagName="h1"
    init={{
      menubar: false,
      inline: true,
      theme: 'inlite',
      content_css: 'css/content.css',
      insert_toolbar: '',
      selection_toolbar: '',
    }}
  />
);

TitleEditor.propTypes = {
  value: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  handlePaste: PropTypes.func.isRequired,
  handleTitlePlaceholderTextFocusIn: PropTypes.func.isRequired,
  handleTitlePlaceholderTextFocusOut: PropTypes.func.isRequired,
};

export default TitleEditor;
