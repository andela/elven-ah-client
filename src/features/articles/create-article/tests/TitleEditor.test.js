import React from 'react';
import { shallow } from 'enzyme';
import { Editor } from '@tinymce/tinymce-react';
import TitleEditor from '../TitleEditor';

const mockFunction = () => jest.fn();

describe('Title Editor component', () => {
  it('should render without errors', () => {
    const wrapper = shallow(<TitleEditor
      handleChange={mockFunction}
      handlePaste={mockFunction}
      value="<h1>This is my Title</p>"
    />);
    expect(wrapper.find(Editor)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
