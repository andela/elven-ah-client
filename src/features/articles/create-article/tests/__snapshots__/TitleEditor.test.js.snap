// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Editor component should render without errors 1`] = `
<Editor
  apiKey="3eytfdm1nnstcd6m2hzat9gpfvgg0ik5im0he4tpolgyqne9"
  init={
    Object {
      "content_css": "css/content.css",
      "inline": true,
      "insert_toolbar": "",
      "menubar": false,
      "selection_toolbar": "",
      "theme": "inlite",
    }
  }
  name="title-editor"
  onEditorChange={[Function]}
  onFocusIn={[Function]}
  onFocusOut={[Function]}
  onPaste={[Function]}
  tagName="h1"
  value="<h1>This is my Title</p>"
/>
`;
